pipelines/cemba/cemba_methylcseq/README.md

```diff
@@ -1,40 +1,186 @@
+| CreateUnmappedBam | [Picard v2.18.23](https://broadinstitute.github.io/picard/)  | Create uBAM for attaching barcodes | quay.io/broadinstitute/picard:2.18.23 |
+| ExtractCellBarcodes | [sctools v0.3.4](https://sctools.readthedocs.io/en/latest/sctools.html)  | Use whitelist to extract barcodes and tag to uBAM | quay.io/humancellatlas/secondary-analysis-sctools:v0.3.4 |
+| Trim | [Cutadapt v1.18](https://cutadapt.readthedocs.io/en/stable/)    | Trim degenerate bases, primer index, C/T Adaptase tail of R1 | quay.io/broadinstitute/cutadapt:1.18 |
+| Trim | [Cutadapt v1.18](https://cutadapt.readthedocs.io/en/stable/)   | Trim bases, primer index, C/T Adaptase tail of R2 | quay.io/broadinstitute/cutadapt:1.18 |
+| Align | [Bismark v0.21.0](https://www.bioinformatics.babraham.ac.uk/projects/bismark/)  | Map multiplexed samples as single-end with --bowtie2 | quay.io/broadinstitute/bismark:0.21.0 |
+| Sort |  [Picard v2.18.23](https://broadinstitute.github.io/picard/) | Sort BAM(s) in coordinate order | quay.io/broadinstitute/picard:2.18.23 |
+| FilterDuplicates |  [Picard v2.18.23](https://broadinstitute.github.io/picard/)  | Removes duplicate reads from BAM | quay.io/broadinstitute/picard:2.18.23 |
+| Get MethylationReport |[Bismark v0.21.0](https://www.bioinformatics.babraham.ac.uk/projects/bismark/)  | Produce methylation report for duplicates-filtered BAM |quay.io/broadinstitute/bismark:0.21.0 |
+| FilterMapQuality | [Samtools v1.9](http://www.htslib.org/)  | Further filter duplicate-removed BAM by map quality | quay.io/broadinstitute/samtools:1.9 |
+| GetMethylationReport | [Bismark v0.21.0](https://www.bioinformatics.babraham.ac.uk/projects/bismark/)  | Produce methylation report for reads above map quality and below map quality | quay.io/broadinstitute/bismark:0.21.0 |
+| AttachBarcodes |  [Picard v2.18.23](https://broadinstitute.github.io/picard/)  | Add barcodes from the tagged uBAM to the aligned BAM | quay.io/broadinstitute/picard:2.18.23 |
+| MergeBams |  [Samtools v.19](http://www.htslib.org/)  | Merge R1 and R2 BAM files into single BAM | quay.io/broadinstitute/samtools:1.9 |
+| AddReadGroup |  [GATK v4.1.2.0](https://gatk.broadinstitute.org/hc/en-us)  | Add read groups to the merged BAM | broadinstitute/gatk:4.1.2.0 |
+| Sort | [Picard v2.18.23](https://broadinstitute.github.io/picard/) | Sort in coordinate order after adding read group | quay.io/broadinstitute/picard:2.18.23 | 
+| IndexBam |  [Samtools v1.9](http://www.htslib.org/)  | Index the output BAM | quay.io/broadinstitute/samtools:1.9 |
+| MethylationTypeCaller | [GATK v4.1.2.0](https://gatk.broadinstitute.org/hc/en-us)  | Produce a  VCF with locus-specific methylation information | broadinstitute/gatk:4.1.2.0 |
+| ComputeCoverageDepth | [Samtools v1.9](http://www.htslib.org/)  | Compute number of sites with coverage greater than 1 | quay.io/broadinstitute/samtools:1.9 |
+
+### Prior to running: set paired-end or single-end mode
```

Comment:
So paired end only works for a single nucleus input? (just want to make sure I understand)

Reply:
Paired-end works for any paired-end input, it just can't be use if you want to actually attach cell barcodes to your reads. I clarified it in the "Prior to running" section.